SimG4CMS/PPS/src/TotemRPSD.cc

```diff
@@ -0,0 +1,295 @@
+// File: TotemRPSD.cc
+// Date: 18.10.2005
+// Description: Sensitive Detector class for TOTEM RP Detectors
+// Modifications:
+#include "SimG4CMS/PPS/interface/TotemRPSD.h"
+#include "SimG4CMS/PPS/interface/PPSStripNumberingScheme.h"
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "SimG4Core/Notification/interface/TrackInformation.h"
+#include "SimG4Core/Notification/interface/G4TrackToParticleID.h"
+#include "SimG4Core/Geometry/interface/SensitiveDetectorCatalog.h"
+#include "SimG4Core/Physics/interface/G4ProcessTypeEnumerator.h"
+
+#include "SimG4Core/SensitiveDetector/interface/SensitiveTkDetector.h"
+
+#include "SimDataFormats/TrackingHit/interface/UpdatablePSimHit.h"
+#include "SimDataFormats/SimHitMaker/interface/TrackingSlaveSD.h"
+
+#include "G4SDManager.hh"
+#include "G4Step.hh"
+#include "G4Track.hh"
+#include "G4VProcess.hh"
+
+#include "G4PhysicalConstants.hh"
+#include "G4SystemOfUnits.hh"
+
+#include <iostream>
+#include <vector>
+#include <string>
+
+TotemRPSD::TotemRPSD(const std::string& name,
+                     const edm::EventSetup& es,
+                     const SensitiveDetectorCatalog& clg,
+                     edm::ParameterSet const& p,
+                     const SimTrackManager* manager)
+    : SensitiveTkDetector(name, es, clg, p),
+      numberingScheme(nullptr),
+      hcID(-1),
+      theHC(nullptr),
+      currentHit(nullptr),
+      theTrack(nullptr),
+      currentPV(nullptr),
+      unitID(0),
+      preStepPoint(nullptr),
+      postStepPoint(nullptr),
+      eventno(0) {
+  collectionName.insert(name);
+
+  edm::ParameterSet m_Anal = p.getParameter<edm::ParameterSet>("TotemRPSD");
+  verbosity_ = m_Anal.getParameter<int>("Verbosity");
+
+  slave = new TrackingSlaveSD(name);
```

Comment:
should be a unique_ptr constructed with make_unique

Reply:
From the discussion below with Vladimir, I understood it should be kept as it is.

Reply:
my understanding of @civanch's comments in https://github.com/cms-sw/cmssw/pull/27673#issue-303505272 is that they only apply to the hit objects. Other objects, including this one, should be converted to unique_ptrs.